src/backend/libc/thread/syscalls.rs

```diff
@@ -299,3 +299,21 @@ pub(crate) fn setns(fd: BorrowedFd, nstype: c::c_int) -> io::Result<c::c_int> {
 pub(crate) fn unshare(flags: crate::thread::UnshareFlags) -> io::Result<()> {
     unsafe { ret(c::unshare(flags.bits() as i32)) }
 }
+
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[inline]
+pub(crate) fn capget(
+    header: *mut linux_raw_sys::general::__user_cap_header_struct,
+    data: *const linux_raw_sys::general::__user_cap_data_struct,
```

Comment:
This function and several like it take raw pointers, so they should be `unsafe fn`.

Also, should `data` here be a `*mut` pointer?

Reply:
> Also, should data here be a *mut pointer?

Really good catch, thank you!

Regarding unsafe, I realized this abstraction sucks anyway: these functions take in references now and are in charge of raw pointer shenanigans.